src/main/java/nextstep/subway/line/domain/PathGraph.java

```diff
@@ -0,0 +1,80 @@
+package nextstep.subway.line.domain;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import nextstep.subway.line.dto.PathResponse;
+import nextstep.subway.station.domain.Station;
+import nextstep.subway.station.domain.StationRepository;
+import nextstep.subway.station.dto.StationResponse;
+import org.jgrapht.GraphPath;
+import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
+import org.jgrapht.graph.DefaultWeightedEdge;
+import org.jgrapht.graph.WeightedMultigraph;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PathGraph {
+
+    private final StationRepository stationRepository;
+
+    public PathGraph(StationRepository stationRepository) {
+        this.stationRepository = stationRepository;
+    }
+
+    public PathResponse findPath(Long source, Long target, List<Line> lines){
+        WeightedMultigraph<String, DefaultWeightedEdge> graph = new WeightedMultigraph(DefaultWeightedEdge.class);
+        DijkstraShortestPath stationGraph = getStationGraph(lines, graph);
+        GraphPath path = stationGraph.getPath(source.toString(), target.toString());
+
+        return new PathResponse(createStations(path), (int)path.getWeight());
+    }
+
+    private DijkstraShortestPath getStationGraph(List<Line> lines, WeightedMultigraph<String, DefaultWeightedEdge> graph) {
+        setVertex(graph, lines);
+        setEdgeWeight(graph, lines);
+        return new DijkstraShortestPath(graph);
+    }
+
+    private void setVertex(WeightedMultigraph<String, DefaultWeightedEdge> graph, List<Line> lines) {
```

Comment:
```suggestion
    private void setVertex(WeightedMultigraph<String, DefaultWeightedEdge> graph, List<Station> stations) {
```
필요한것만 파라미터로 받는건 어때요?